tests/configlet/util/generic_patch.py

```diff
@@ -0,0 +1,217 @@
+#! /usr/bin/env python
+
+import argparse
+import json
+import jsonpatch
+import fnmatch
+import os
+import re
+import sys
+import time
+import yaml
+
+from helpers import *
+from common import *
+
+if os.path.exists("/etc/sonic/sonic-environment"):
+    from mock_for_switch import config_reload, wait_until
+else:
+    from tests.common.config_reload import config_reload
+    from tests.common.utilities import wait_until
+
+CMD_APPLY_HACK = "config apply-patch -n -i '' -v {}"
+CMD_APPLY = "config apply-patch -n -i /FEATURE -i /SCHEDULER -i /QUEUE -i /CABLE_LENGTH -i /ACL_TABLE/E -v {}"
+
+# HACKS summary:
+#
+# 1) While adding i/f (add_t0), interfaces must be added at the start, before
```

Comment:
Shouldn't it be enough to only fix `2)` which will handle `1)`? `2)` was fixed in https://github.com/Azure/sonic-utilities/issues/2007

Reply:
Maybe I can test it out

Reply:
I tried deleting the `INTERFACE` table, that caused the BGP_NEIGHBORS to be `Idle` then `Active`. Then I added the `INTERFACE` back and that caused the BGP_NEIGHBORS to be brought back up.

Here is the output (running using T2 topology, for some reason I can't deploy T1 topology on my ubuntu vm):
```sh
admin@vlab-t2-02:~$ show ip bgp summary
.
.
.
10.0.0.11      4  65000        571        571         0      0       0  00:27:32               65  ARISTA03T1
.
.
.
admin@vlab-t2-02:~$ sudo config apply-patch remove-interface.json -i ''
Patch Applier: Patch application starting.
Patch Applier: Patch: [{"op": "remove", "path": "/INTERFACE"}]
.
.
.
Patch applied successfully.
admin@vlab-t2-02:~$ show ip bgp summary
.
.
.
10.0.0.11      4  65000        575        593         0      0       0  00:27:48               65  ARISTA03T1
.
.
.
admin@vlab-t2-02:~$ show ip bgp summary
.
.
.
10.0.0.11      4  65000        575        597         0      0       0  00:00:00   Idle            ARISTA03T1
.
.
.
admin@vlab-t2-02:~$ show ip bgp summary
.
.
.
10.0.0.11      4  65000        575        597         0      0       0  00:00:04   Active          ARISTA03T1
.
.
.
admin@vlab-t2-02:~$ sudo config rollback cp1 -i ''
Config Rollbacker: Config rollbacking starting.
.
.
.
Patch Applier: Patch: [{"op": "add", "path": "/INTERFACE", "value": {"Ethernet8": {}, "Ethernet8|10.0.0.10/31": {}, "Ethernet8|FC00::15/126": {}}}]
.
.
.
Config rolled back successfully.
admin@vlab-t2-02:~$ show ip bgp summary
.
.
.
10.0.0.11      4  65000        599        621         0      0       0  00:00:11               65  ARISTA03T1
.
.
.
admin@vlab-t2-02:~$ 
```

Reply:
In RemoveRack, it is more than BGP neighbor. You remove the port settings, like BUFFER, QOS, .... The order has to be remove those first, then remove the port. BGP is at a higher level in the the networking stack, that it does not directly deal with i/f but traffic. Hence as long as port is marked down at the start, the BGP goes down until the traffic flows again, which will be only for AddRack upon port going up.

So we do need 1

Reply:
OK, I think I have to use RemoveRack scenario for testing. I have created a ticket for this and will follow-up there. https://github.com/Azure/sonic-utilities/issues/2051